### DISCUSSION TITLE
autoRefreshToken: false still auto refreshes. What am I missing?

### DISCUSSION THREAD

**Discussion by rodpatulski** [supabase/supabase#17788] [Questions]
So i'm logging in in a react app using
`export const supabase = createClient<Database>(supabaseUrl, supabaseAnonKey,{
	auth:{
		autoRefreshToken: false
	}
})`

I see an access token and refresh token stored in my local storage. All is good. 
I set my jwt_expiry = 5 in config.toml so that the jwt expires after 5 seconds for testing.

However, my access token and refresh token, keep refreshing automatically every 5 seconds without me calling getSession() or any other supabase function.

What am I missing here? Are refresh token living forever by design? If so, why is there an option for autoRefreshToken?

Help y'all.

Thanks

**Comment by GaryAustin1** [comment]
The refresh token is good until used.
The expiry is just the access token expires. 

Auto refresh just means it will refresh the jwt (using the refresh token)  with a timer before it expires.

The session will still refresh if you signin again and have persistSession:true.

Also you should not set jwt_expiry less than a minute or so IMO.  There used to be issues with supabase-js if the expire time was less than 120.  I've not tested lately to see if those issues are gone or not.

**Comment by DeveloperSensorberg** [comment]
How does this make sense, autoRefreshToken is false and I still see the token getting refreshed either way with the auth change listener

**Reply by GaryAustin1** [reply]
Show your createClient call.  Also it will get refreshed when it expires and you call getSession.

**Comment by singh-inder** [comment]
Hi, I'm stuck on the same problem. Is there any solution to this? 

Here is my code. Pretty much same as OP.
```ts
      const client = createClient(supabaseUrl, supabaseAnonKey, {
        auth: {
          autoRefreshToken: false
        }
      });
```

**Reply by GaryAustin1** [reply]
I believe that setting just impacts the timer the clients use to refresh the access_token with the one time refresh token.  Starting createClient will refresh the token if expired, persistSession flag would impact that.  getSession also refreshes the the access_token.

**Reply by rodpatulski** [reply]
The thing is that I would expect both tokens to not change when autorefreshtoken is false. I'm literally sitting in the browser watching both of them change every 5 seconds without me logging in or calling any supabase functions. So what is the point of autorefreshtoken? This is still a mystery to me.

**Reply by GaryAustin1** [reply]
Not sure it is related but SB recommends no less that 5 minutes: https://supabase.com/docs/guides/auth/sessions
BUT,  I just set my supabase-js app to not autoRefreshToken:false and set the jwt to 5 seconds.

No refreshes and my operations fail with jwt expired.

With autoRefreshToken not supplied, I see constant refreshes.

**Reply by Justinyu1618** [reply]
FYI, I ran into a similar issue if you use the `createBrowserClient` function from `supabase/ssr`

Submitted an [issue](https://github.com/supabase/ssr/issues/108)
